### PR TITLE
Change unit.status type to enum

### PR DESCRIPTION
**Pull request by abbiefarr**
## Description

Creates a UnitStatus enum and changes the type of unit.status from string to UnitStatus.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [x] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

I launched the public portal and was able to see "Units Available". I also launched the partner portal and when through both the add and edit listings workflows and they worked fine.

NOTE: editing the unit status is disabled in the ui, so that will likely need to be revisited when it's enabled.

Please describe the tests that you ran to verify your changes. Provide instructions so we can review. Please also list any relevant details for your test configuration
- [x] Desktop View
- [ ] Mobile View

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes

### CODE REVIEW

File: backend/core/src/migration/1626207809474-unit-status-enum.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+
+export class unitStatusEnum1626207809474 implements MigrationInterface {
+    name = 'unitStatusEnum1626207809474'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`ALTER TABLE "units" DROP COLUMN "status"`);
```
Comment (pbn4) [inline backend/core/src/migration/1626207809474-unit-status-enum.ts:7]: This migration will result in data loss in existing DBs, please first create a new type, then use alter table / alter column as e.g. [here](https://stackoverflow.com/a/15655926/2429333)

**Review by pbn4 — CHANGES_REQUESTED** [review]

File: backend/core/src/migration/1626207809474-unit-status-enum.ts (hunk 1 above)
Comment (abbiefarr) [reply backend/core/src/migration/1626207809474-unit-status-enum.ts]: Good catch, thanks! Is the most recent version what you're looking for?

Comment (pbn4) [reply backend/core/src/migration/1626207809474-unit-status-enum.ts]: Yes, thank you.

**Comment by pbn4** [comment]
@abbiefarr I closed/opened to retrigger CI/CD

**Comment by pbn4** [comment]
@abbiefarr Tests failed because environment variables are missing, the problem is the fork coming from a 3rd party organization - our CI/CD is not setting secrets for such builds. And since there is no granular control of who we can share it with  (we either share with everyone of with noone) we decided not to do this. It means that `.circleci/config.yml` requires fixes and providing some mock env variables builds are complaining about. I can fix it in your branch tomorrow morning CET.

**Comment by abbiefarr** [comment]
@pbn4 sounds good, thanks!

**Comment by pbn4** [comment]
@abbiefarr I have taken over in https://github.com/bloom-housing/bloom/pull/1526 because I have no permissions to commit to your fork, but unfortunately I didn't have enough time to fix cypress/run today. @seanmalbert @software-project  Could you take over? I gave you all permissions to my fork

**Comment by pbn4** [comment]
@abbiefarr Please merge latest dev and check if it works for you now.

**Comment by seanmalbert** [comment]
@abbiefarr  and @pbn4 ,
I resolved the conflicts and committed to a branch on my own fork (I don't have permissions to push to detroit). The PR is at #1556 . So I'm going to close this and use that.

**Pull request closed without merging**